sdks/java/io/hadoop-input-format/src/test/java/org/apache/beam/sdk/io/hadoop/inputformat/unit/tests/inputformats/BadCreateReaderEmpInputFormat.java

```diff
@@ -0,0 +1,179 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.beam.sdk.io.hadoop.inputformat.unit.tests.inputformats;
+
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+import org.apache.hadoop.io.Text;
+import org.apache.hadoop.io.Writable;
+import org.apache.hadoop.mapreduce.InputFormat;
+import org.apache.hadoop.mapreduce.InputSplit;
+import org.apache.hadoop.mapreduce.JobContext;
+import org.apache.hadoop.mapreduce.RecordReader;
+import org.apache.hadoop.mapreduce.TaskAttemptContext;
+
+/**
+ * <p>This is InputFormat for reading employee data which is stored in map employeeData. employeeData
+ * has 9 employee records. 
+ * <p>EmployeeInputFormat splits data into 3 splits each having 3 records.
+ * <p>EmployeeInputFormat reads data from employeeData and produces a key is of type Text which is employee id and
+ * value of type {@link org.apache.beam.sdk.io.hadoop.inputformat.unit.tests.inputformats.Employee
+ * Employee}.
+ * <p>This is test input to validate HadoopInputFormatIO stops reading if createRecordReader throws exception.
+ */
+public class BadCreateReaderEmpInputFormat extends InputFormat<Text, Employee> {
+  private static final long numberOfRecordsInEachSplit = 3L;
+  private static final long numberOfSplits = 3L;
+
+  public BadCreateReaderEmpInputFormat() {}
+
+  @Override
+  public RecordReader<Text, Employee> createRecordReader(InputSplit split,
+      TaskAttemptContext context) throws IOException, InterruptedException {
+    return new BadCreateRecordRecordReader();
+  }
+
+  @Override
+  public List<InputSplit> getSplits(JobContext jobContext)
+      throws IOException, InterruptedException {
+    List<InputSplit> inputSplitList = new ArrayList<InputSplit>();
+    for (long i = 0L; i < numberOfSplits; i++) {
+      InputSplit inputSplit = new BadCreateRecordReaderInputSplit((i * numberOfSplits),
```

Comment:
same error I noticed elsewhere? should this be numberOfSplits -> numberOfRecordsInEachSplit